pkg/aws/aws.go

```diff
@@ -215,6 +223,23 @@ func GetDefaultAMI(ctx context.Context, cfg aws.Config, instanceType string) (st
 	return *result.Images[0].ImageId, nil
 }
 
+func GetAMIRootDevice(ctx context.Context, cfg aws.Config, diskImage string) (string, error) {
+	svc := ec2.NewFromConfig(cfg)
+
+	input := &ec2.DescribeImagesInput{
+		ImageIds: []string{
+			diskImage,
+		},
+	}
+	result, err := svc.DescribeImages(ctx, input)
+
+	if err != nil {
+		return "", err
+	}
+
+	return *result.Images[0].RootDeviceName, nil
```

Comment:
would be great if you could check these values for `nil`s

Reply:
672ee4a